pkgs/by-name/at/atlauncher/package.nix

```diff
@@ -1,52 +1,66 @@
-{ copyDesktopItems, fetchurl, jre, lib, makeDesktopItem, makeWrapper, stdenv, udev, xorg }:
+{ copyDesktopItems
+, fetchFromGitHub
+, fetchurl
+, lib
+, makeDesktopItem
+, makeWrapper
+, stdenvNoCC
 
-stdenv.mkDerivation (finalAttrs: {
+, # dependencies
+  jre
+, udev
+, xorg
+}:
+
+let
   pname = "atlauncher";
-  version = "3.4.35.4";
+  version = "3.4.35.9";
 
-  src = fetchurl {
-    url = "https://github.com/ATLauncher/ATLauncher/releases/download/v${finalAttrs.version}/ATLauncher-${finalAttrs.version}.jar";
-    hash = "sha256-M8ygN70yizJM6VEffBh/lH/DneKAzQ5UFzc3g51dja0=";
+  srcs = {
+    source = (fetchFromGitHub {
+      owner = "ATLauncher";
+      repo = "ATLauncher";
+      rev = "v${version}";
+      hash = "sha256-QbSCerrALrbHMWuPgJEthUrcYlaDEWuU6Kzwqe2mr8Q=";
+    });
+    jar = (fetchurl {
+      url = "https://github.com/ATLauncher/ATLauncher/releases/download/v${version}/ATLauncher-${version}.jar";
+      hash = "sha256-Y2MGhzq4IbtjEG+CER+FWU8CY+hn5ehjMOcP02zIsR4=";
+    });
   };
```

Comment:
why not use `srcs` of the derivation itself? This makes it very hard to override this derivation

Reply:
I can't seem to grab either of the sources deterministically in that case, i.e., `$srcs[0]` can either be the GH source or the JAR; and abstracting just the fetches and then doing `srcs = [ src jar ];` doesn't solve the point you raise.

Reply:
You *should* probably be using relative paths in the working directory of the phases to reach the files you want. All derivations defined in `srcs` get symlinked to the build environment

Reply:
That's correct; I've seen other packages do similar; but they either `cp` everything, or fetch the stuff into a variable prior and use `${some-src.name}/` to access its directory.

When I try running `ls` within the `installPhase`, I just get `env-vars/` which I assume is just the env being setup in that dir. Not sure how to access the sources there without doing `srcs[0]` which is inconsistent in my case.